client/app/src/main/java/com/microsoft/research/karya/data/model/karya/enums/RegisterWorkerState.kt

```diff
@@ -0,0 +1,7 @@
+package com.microsoft.research.karya.data.model.karya.enums
+
+enum class RegisterWorkerState {
+    NOT_STARTED,
+    SUCCESS,
+    FAILURE
+}
```

Comment:
We should probably make this a `sealed class` and pass relevant data along with the state.

Reply:
This is a really good idea